docs/sdg/sdg-flow-yaml.md

```diff
@@ -0,0 +1,205 @@
+
+We will continue to use these config files unchanged, and custom files can be specified with an absolute path. Relative paths are assumed to be relative to the Python package e.g. `configs/skills/...`.
+
+### Model Serving
+
+Custom pipelines may have more unique model serving requirements. Instead of serving just one model, we may need to launch the model server with a model and an additional model with adapter. vLLM, for example, can host both a model and a model+adapter under two different model IDs.
+
+The pipeline author needs some way of disambiguating between these multiple models - i.e. the definition of each `LLMBlock` needs to specify a particular model.
+
+Right now the `Pipeline` constructor takes two relevant parameters - the OpenAI client instance, and the model ID for the default model. It's important to note that this model ID is defined by the user at runtime, and it may not match the model IDs that the pipeline author used.
+
+The use cases will be:
+
+1. Most LLMBlock definitions will use the default teacher model - and we can make the semantic that if the pipeline author doesn't specify a model in an `LLMBlock`, the default in `PipelineContext.model_id` is used.
+2. In cases where a model+adapter is to be served, the pipeline author should choose a descriptive model ID using `block.gen_kwargs.model_id` and the user should ensure that this is the model ID that is served.
+
+For example, a pipeline author might define:
+
+```yaml
+version: "1.0"
+blocks:
+  - name: gen_questions
+    type: LLMBlock
+    config:
+      config_path: configs/skills/freeform_questions.yaml
+      add_num_samples: True
+      gen_kwargs:
+        model_id: mycustomadapter
+      output_cols:
+        - question
+    drop_duplicates:
+      - question
+```
+
+and the user will be required to define a serving configuration like:
+
+```bash
+--lora-modules=mycustomadapter=path/to/my_custom_adapter
+```
+
+### CLI Integration
+
+As of the current version of `ilab`, it supports `simple` and `full` as parameters to `--pipeline` to select one of the two types of built-in pipelines included in the library.
+
+Once we have support for loading custom pipelines, we need a way for these to be specified with the CLI. We believe the most common case for custom pipelines is for them to extend the `full` pipeline and, as such, we should support extending existing pipelines with a custom pipeline rather than simply specifiying a single pipeline.
+
+Here is a proposed CLI UX for this:
+
+> `ilab data generate`
+
+Use the default pipeline, `simple`.
+
+> `ilab data generate --pipeline full`
+
+Use the built-in `full` pipeline.
+
+> `ilab data generate --pipeline full --pipeline path/to/config.yaml`
+
+Use the `full` pipeline, but extend it with the pipeline defined in the provided config file.
+
+> `ilab data generate --pipeline path/to/other_config.yaml`
+
+Do not use a built-in pipeline. Use only the pipeline in the custom configuration.
```

Comment:
related to https://github.com/instructlab/dev-docs/pull/109#discussion_r1674537552

i think the relationships between existing and custom flows are too complicated to express in terms of command line args.  I think we're going to have to find a way to express it in the files themselves for custom flows. 

Also related to this discussion where our custom pipeline configuration is getting complex enough to need to be a directory of files: https://github.com/instructlab/dev-docs/pull/109/files#r1673815390

With these points, I think we can simplify the CLI integration back down to:

```suggestion
> `ilab data generate --pipeline full`

Use the built-in `full` pipeline.

> `ilab data generate --pipeline path/to/custom_pipeline_directory/`

Use a custom pipeline configuration. The custom pipeline may include references to the built-in flows to be used in conjunction with custom ones, but those details are contained within the yaml files in the custom directory.
```

Reply:
Sounds good to me :+1:

Reply:
Nice. This 100% aligns what I think :D

Reply:
I applied the proposed text here that simplified the CLI integration proposed